mixer/adapter/prometheus/prometheus.go

```diff
@@ -188,7 +212,15 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 		return nil, err
 	}
 
-	return &handler{b.srv, b.metrics}, metricErr.ErrorOrNil()
+	var expiryCache cache.ExpiringCache
+	if cfg.MetricsExpirationPolicy != nil {
+		expiryCache = cache.NewTTLWithCallback(
+			cfg.MetricsExpirationPolicy.MetricsExpiryDuration,
```

Comment:
Does the go routine spawned by cache dies once there is no reference to the cache?
Is there a finalizer ? I think this will happen when a new handler is created.

Reply:
There is a finalizer in the ttl cache impl. It sends a message on a stop channel and waits for the evicting goroutine to stop.